### DISCUSSION TITLE
Zed running on FreeBSD!

### DISCUSSION THREAD

**Discussion by syobocat** [zed-industries/zed#10247] [Show and tell]
Zed runs on FreeBSD, with small modifications!
![image](https://github.com/zed-industries/zed/assets/46315199/94ca7b13-c98c-4224-96d3-81b8c84521cf)

Here's what I did:
- Replaced every `#[cfg(target_os = "linux")]` with `#[cfg(any(target_os = "linux", target_os = "freebsd"))]` (I found them on `gpui` and `blade-graphics`)
- Modified `ipc-channel` for FreeBSD libc (I don't know anything about libc, so I just followed hints which Cargo provided)
- Added `-C link-dead-code` to `RUSTFLAGS`

I'm not a developer or anything like that and I just did it out of curiosity, but I hope this helps other *BSD users to run Zed on their system, or Zed developers to support more platforms!

**Comment by AlekseyZaytcev** [comment]
Can you share your impressions? Is it ready for work at current state?

**Comment by ARPABoy** [comment]
It would be great FreeBSD ready source code and binary releases.

**Comment by rudrabhoj** [comment]
I tried to compile it on FreeBSD 14.1, replaced as you asked "Replaced every #[cfg(target_os = "linux")] with #[cfg(any(target_os = "linux", target_os = "freebsd"))]". But I get a lot of these errors:

use of undeclared crate or module `cosmic_text`

You can see the entire error here: https://pastebin.com/Wnj6vCA3

**Comment by mykola2312** [comment]
Can you please elaborate? Provide patches?

**Comment by mykola2312** [comment]
@syobocat

**Comment by syobocat** [comment]
Well, I haven't followed up after the post, so I can't say anything right now.
I think I'll try this again when I have time.

**Comment by chiefMarlin** [comment]
Please keep us posted, would be great to have an official freebsd release!

**Comment by syobocat** [comment]
#20480
👀
